python/hail/matrixtable.py

```diff
@@ -2878,14 +2889,16 @@ def _select_rows(self, caller, key_struct=None, value_struct=None, pk_size=None)
         analyze(caller, row, self._row_indices, {self._col_axis})
         base, cleanup = self._process_joins(row)
 
-        return cleanup(MatrixTable(base._jvds.selectRows(row._ast.to_hql(), new_key)))
+        return cleanup(MatrixTable(base._jvds.selectRows(str(row._ir), new_key)))
 
-    @typecheck_method(key_struct=nullable(expr_struct()),
-                      value_struct=nullable(expr_struct()),
-                      pk_size=nullable(int))
-    def _select_rows_processed(self, key_struct=None, value_struct=None, pk_size=None):
-        row, new_key = self._make_row(key_struct, value_struct, pk_size)
-        return MatrixTable(self._jvds.selectRows(row._ast.to_hql(), new_key))
+    @typecheck_method(key_struct=expr_struct())
+    def _select_rows_processed(self, key_struct):
```

Comment:
from the code changes, it looks like this argument is the value struct, not the key struct. Is that right?

Reply:
I'm pretty sure the only argument that's being used is the key struct---it's used to select the groups for grouped matrix tables.